### ISSUE TITLE
GLM Tweedie Distribution and Identity Link 

### ISSUE THREAD

**Issue by ajiang2021** [statsmodels/statsmodels#7328]
#### Describe the bug

While exploring Statsmodels GLM functionalities, I noticed that I'm able to fit a GLM with a Tweedie distribution and identity link. However I noticed that on the documentation page identity link is not one of the links listed for Tweedie https://www.statsmodels.org/stable/generated/statsmodels.genmod.families.family.Tweedie.html) and I get the same conclusion if I run sm.families.family.Tweedie.links.

Is the Tweedie distribution and identity link combination currently supported? If so, could this be updated in the documentation? If not, could this be a bug and this combination should have errored out when it was being passed to the training code? 

#### code sample
import numpy as np
import statsmodels
import statsmodels.api as sm 

#create predictor dataset 
np.random.seed(0)
n=100000
p=3
X = np.random.rand(n,p) 

#coefficients
beta = [1,2,1]

#generate target 
mu = np.matmul(X,beta)
Y = np.random.poisson(lam=mu)

#fit model using Tweedie distribution and identity link
model=sm.GLM(endog=Y, exog=X, family=sm.families.Tweedie(link=sm.families.links.identity()))
fitted_model = model.fit()
fitted_model.summary()

#### Expected Output
If Tweedie and identity link combination is not supported, I would have expected this to error out instead of completing the training. If it is, it would be great if this could be added to the documentation so users know this combination is available for use.

**Comment by josef-pkt** [comment]
The link functions impose constraints that the parameters are in a valid range, e.g. positive intensities in Poisson and probabilities in (0, 1).

In some cases those constraints are satisfied with other links without explicitly imposing them. However, in general models and for general data, the constraints will often be violated especially during the optimization, and the algorithm breaks down or produces invalid predictions, e.g. negative probabilities.

So identity link will work in some cases but will often not work, and we only warn instead of raising an exception.

For example: When I estimate a one sample or two sample problem with GLM, I often use identity link to check the mean(s) directly. With good starting values, this always works (except at boundaries).

**Comment by jmahlik** [comment]
See https://github.com/statsmodels/statsmodels/pull/7395.  I ran in to the same confusion where `identity` isn't explicitly called out but is just `Power(1)` (alias of power). Updated the docs a tad to hopefully point folks in the right direction. Might be ok to close this?

**Comment by josef-pkt** [comment]
related FAQ issue #2208  but it's mostly a discussion when we figured this out the first time.
my comment here is a good summary
